File: benchmarks/tasks/word_freq/word_freq.hs
-- SPDX-License-Identifier: MIT-0
module Main where

count :: String -> [String] -> Int
count w = length . filter (== w)

main :: IO ()
main = do
  let ws = words "the cat sat on the mat the cat"
  putStrLn ("the: " ++ show (count "the" ws))
  putStrLn ("cat: " ++ show (count "cat" ws))
  putStrLn ("sat: " ++ show (count "sat" ws))

File: benchmarks/tasks/word_freq/word_freq.py
# SPDX-License-Identifier: MIT-0

words = "the cat sat on the mat the cat".split()

def count(w, ws):
    return sum(1 for x in ws if x == w)

print(f"the: {count('the', words)}")
print(f"cat: {count('cat', words)}")
print(f"sat: {count('sat', words)}")
